jerry-core/vm/vm.c

```diff
@@ -96,7 +96,16 @@ vm_op_get_value (ecma_value_t object, /**< base object */
 
   if (unlikely (ecma_is_value_undefined (object) || ecma_is_value_null (object)))
   {
-    return ecma_raise_type_error (ECMA_ERR_MSG ("Base object cannot be null or undefined."));
+    ecma_string_t *property_name_p = ecma_get_string_from_value (property);
+    ECMA_STRING_TO_UTF8_STRING (property_name_p, property_name_utf8_p, property_name_size);
```

Comment:
Personally, I would prefer if the messages were created using the string helper functions found in `ecma-helpers-string.c`, instead of using intermediate buffers and snprintf, if it's possible. Maybe like defining a few magic strings, and using `ecma_concat_ecma_strings` to create the final message.